src/admin/class-helper.php

```diff
@@ -54,4 +56,165 @@ function () use ( $message ) {
 			}
 		);
 	}
+
+	/**
+	 * Get the feature collection from a datalayer.
+	 *
+	 * @param int $datalayer_id The datalayer ID.
+	 *
+	 * @return string The feature collection.
+	 */
+	public static function get_feature_collection_from_datalayer( $datalayer_id ) {
+		global $wpdb;
+
+		$datalayer = get_post( $datalayer_id );
+
+		// Create a custom query to get the location ID's, order by title ASC, where the geometry exists and the location_datalayer_id is equal to the object ID.
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching -- Custom query is required here.
+		$datalayer_locations_db = $wpdb->get_results(
+			$wpdb->prepare(
+				"SELECT p.ID FROM {$wpdb->posts} p
+			LEFT JOIN {$wpdb->postmeta} pm ON p.ID = pm.post_id
+			LEFT JOIN {$wpdb->postmeta} pm2 ON p.ID = pm2.post_id
+            WHERE p.post_type = 'owc_ok_location'
+              AND p.post_status = 'publish'
+              AND pm.meta_key = 'geometry'
+              AND pm.meta_value IS NOT NULL
+              AND pm2.meta_key = 'location_datalayer_id'
+              AND pm2.meta_value = %s
+            ORDER BY p.post_title ASC",
+				$datalayer_id
+			)
+		);
+
+		$datalayer_locations = [];
+
+		foreach ( $datalayer_locations_db as $location_db ) {
+			$location = self::prepare_location( $location_db->ID );
```

Comment:
Je doet deze `prepare_location` nu elke keer dat deze datalayer wordt opgevraagd, waarom niet voor 95% gewoon opslaan bij het wijzigen van de datalayer? Als ik die functie bekijk zie ik namelijk volgens mij maar één ding wat je echt on-the-fly MOET doen en dat is als er een andere projection gevraagd wordt. Alle andere zaken liggen toch vast op het moment van import? Of zie ik dat verkeerd?
Dan zou je in het geval van de standaard projection de location één-op-één kunnen overnemen, zonder daar nog een hele functie overheen te laten gaan. Door bij het opslaan er voor te zorgen dat alle locations met de meest gebruikte projection worden opgeslagen, zorg je er voor dat je zo min mogelijk on-the-fly hoeft te doen en dus de endpoints sneller zijn.